rest/data/pod.go

```diff
@@ -42,3 +83,48 @@ func (c *MockPodConnector) CheckPodSecret(id, secret string) error {
 	}
 	return errors.New("pod does not exist")
 }
+
+func validatePod(p restModel.APICreatePod) (*pod.Pod, error) {
+	if p.Image == nil {
```

Comment:
I would move the request input validation into `Parse` (along with also unmarshalling the request body in `Parse`).